tests/helpers/perf_run_helper.rb

```diff
@@ -27,6 +28,190 @@ def perf_teardown
+      File.write("#{scenarios_dir}/#{scenario}", json.to_json)
+
+      # update the data for the next iteration
+      @scale_instances = @scale_instances + @scale_increment
+      desc = "'role::by_size_small' role from perf control repo, #{@scale_instances} agents, 1 iteration"
+      json["run_description"] = desc
+      json["nodes"][0]["num_instances"] = @scale_instances
+    end
+
+    # upload scenarios to metrics
+    scp_to(metric, "#{scenarios_dir}", "gatling-puppet-load-test/simulation-runner/config")
+  end
+
+  def check_scale_results(scenario)
+    result_dir = "root/gatling-puppet-load-test/simulation-runner/results"
+    global_stats = "js/global_stats.json"
+
+    # stop monitoring and get results
+    puts "Generating results for scenario: #{scenario}"
+    puts
+
+    atop_result, gatling_result = get_perf_result
+
+    # create scale result folders
+    scale_result_dir = "PERF_SCALE_#{@scale_timestamp}/#{scenario.gsub(".json", "")}"
+    FileUtils.mkdir_p scale_result_dir
+
+    # copy entire results to scale results dir (TODO: remove this?)
+    FileUtils.copy_entry @archive_root, "#{scale_result_dir}/#{@archive_root}"
+
+    # copy metric
+    metric_results = "#{@archive_root}/#{metric.hostname}/#{result_dir}/#{@dir_name}"
+    FileUtils.copy_entry metric_results, "#{scale_result_dir}/metric"
+
+    # copy master
+    master_results = "#{@archive_root}/#{master.hostname}"
+    FileUtils.copy_entry master_results, "#{scale_result_dir}/master"
+
+    # copy stats
+    stats_path = "#{scale_result_dir}/metric/#{global_stats}"
+    json_dir = "PERF_SCALE_#{@scale_timestamp}/json"
+    FileUtils.mkdir_p json_dir
+    FileUtils.copy_file(stats_path, "#{json_dir}/#{scenario}")
+
+    # check the results for KOs (TODO: needed or just use assertion?)
+    puts "Checking results: #{stats_path}"
+    puts
+
+    stats_file = File.read(stats_path)
+    stats_json = JSON.parse(stats_file)
+
+    num_total = stats_json["numberOfRequests"]["total"]
+    num_ok = stats_json["numberOfRequests"]["ok"]
+    num_ko = stats_json["numberOfRequests"]["ko"]
+    puts "Number of requests:"
+    puts "total: #{num_total}"
+    puts "ok: #{num_ok}"
+    puts "ko: #{num_ko}"
+    puts
+
+    if num_ko != 0
+      fail_test "ERROR: KO encountered in scenario: #{scenario}"
+    end
+
+    scale_assertions(atop_result, gatling_result)
+  end
+
+  def scale_assertions(atop_result, gatling_result)
+    step 'max response' do
+      # TODO: enable?
+      # Temporarily disabling this step until SLV-208 is complete
+      # assert_later(gatling_result.max_response_time_agent <= 20000, "Max response time per agent run was: #{gatling_result.max_response_time_agent}, expected <= 20000")
+    end
+
+    step 'successful request percentage' do
+      assert_later(gatling_result.successful_requests == 100, "Total successful request percentage was: #{gatling_result.successful_requests}%, expected 100%" )
+    end
+
+    step 'average memory' do
```

Comment:
just remove these steps that are commented out...

Reply:
Removed